GitUI/CommandsDialogs/BrowseDialog/FormUpdates.cs

```diff
@@ -145,11 +148,27 @@ private void linkChangeLog_LinkClicked(object sender, LinkLabelLinkClickedEventA
             Process.Start("https://github.com/gitextensions/gitextensions/blob/master/GitUI/Resources/ChangeLog.md");
         }
 
-        private void btnDownloadNow_Click(object sender, EventArgs e)
+        private async void btnDownloadNow_Click(object sender, EventArgs e)
         {
             try
             {
-                Process.Start(UpdateUrl);
+                progressBar1.Visible = true;
+                btnDownloadNow.Enabled = false;
+                UpdateLabel.Text = _downloadingUpdate.Text;
+
+                WebClient webClient = new WebClient();
+                string fileName = UpdateUrl.Substring(UpdateUrl.LastIndexOf("/") + 1);
```

Comment:
Looks like we can get away with: 
```suggestion
                string fileName = Path.GetFileName(UpdateUrl);
```
![image](https://user-images.githubusercontent.com/4403806/58554214-bf053300-8259-11e9-9d38-4dc41de658fd.png)

Reply:
Handy, I'll remember this one! I'll update that as well :)

Reply:
Done